### PR TITLE
[CALCITE-5688] Support TRUNCATE TABLE DDL statement in server module

### PR DESCRIPTION
**Pull request by JiajunBernoulli**
Close #3188

### CODE REVIEW

File: babel/src/main/java/org/apache/calcite/sql/babel/SqlTruncateTable.java
Comment (ILuffZhe) [inline babel/src/main/java/org/apache/calcite/sql/babel/SqlTruncateTable.java:37]: What if we extend one SqlKind like `TRUNCATE_TABLE` for this?
I noticed that `OTHER_DDL` is for short term usage in the javadoc.

**Review by ILuffZhe — COMMENTED** [review]
Looks good! Just one minor question.

File: babel/src/main/java/org/apache/calcite/sql/babel/SqlTruncateTable.java
Comment (JiajunBernoulli) [reply babel/src/main/java/org/apache/calcite/sql/babel/SqlTruncateTable.java:37]: Because I don't want to break the core module for now.

Comment (JiajunBernoulli) [reply babel/src/main/java/org/apache/calcite/sql/babel/SqlTruncateTable.java:37]: Let us discuss in JIRA: https://issues.apache.org/jira/browse/CALCITE-5688

File: core/src/main/java/org/apache/calcite/sql/SqlTruncate.java
Comment (ILuffZhe) [inline core/src/main/java/org/apache/calcite/sql/SqlTruncate.java:25]: I have one small question, except `TRUNCATE TABLE`, is there any other truncate syntax in the SQL standard?

File: core/src/main/java/org/apache/calcite/sql/ddl/SqlTruncateTable.java
Comment (ILuffZhe) [inline core/src/main/java/org/apache/calcite/sql/ddl/SqlTruncateTable.java:33]: Nitpick: maybe `Parses` is better based on the following java doc. It's your call.

File: server/src/test/resources/sql/table.iq
Comment (ILuffZhe) [inline server/src/test/resources/sql/table.iq:218]: Can we have a positive test here? To tell the effectiveness of TRUNCATE

File: core/src/main/java/org/apache/calcite/sql/SqlTruncate.java
Comment (JiajunBernoulli) [reply core/src/main/java/org/apache/calcite/sql/SqlTruncate.java:25]: As far as I know, there's nothing else.

The abstract class is reserved for the future.

File: server/src/test/resources/sql/table.iq
Comment (JiajunBernoulli) [inline server/src/test/resources/sql/table.iq:54]: `TRUNCATE` is DDL like `DROP`, so `0 rows modified`.

File: core/src/main/java/org/apache/calcite/sql/ddl/SqlTruncateTable.java
Comment (JiajunBernoulli) [reply core/src/main/java/org/apache/calcite/sql/ddl/SqlTruncateTable.java:33]: I like to maintain a unified style.😊
image

File: server/src/test/resources/sql/table.iq
Comment (JiajunBernoulli) [reply server/src/test/resources/sql/table.iq:233]: I use more efficient methods to clear data and add one positive test.

File: server/src/main/java/org/apache/calcite/server/ServerDdlExecutor.java
Comment (ILuffZhe) [inline server/src/main/java/org/apache/calcite/server/ServerDdlExecutor.java:198]: May I ask why only MutableArrayTable is supported? ^_^

**Review by ILuffZhe — COMMENTED** [review]
I think the PR is in a good shape, just two minor comments.

File: server/src/test/resources/sql/table.iq
Comment (ILuffZhe) [reply server/src/test/resources/sql/table.iq:54]: Good to know, I also tried in MySQL and confirmed your statement.

File: server/src/main/java/org/apache/calcite/server/ServerDdlExecutor.java
Comment (JiajunBernoulli) [reply server/src/main/java/org/apache/calcite/server/ServerDdlExecutor.java:198]: - The DELETE statement cannot delete data in the server module.
   I initially thought it was a [bug](https://issues.apache.org/jira/browse/CALCITE-5796). But after analysis, it  is limitation of the server module.
- `MutableArrayTable` is enough, because `CREATE` table all are `MutableArrayTable`. Here is code link: https://github.com/apache/calcite/blob/301f770a66ef3d26c9d4e8d274eae0f92a819e00/server/src/main/java/org/apache/calcite/server/ServerDdlExecutor.java#L515

Comment (ILuffZhe) [reply server/src/main/java/org/apache/calcite/server/ServerDdlExecutor.java:198]: OK, got it. Maybe some java doc is better for other developers, WDYT?

File: core/src/main/codegen/templates/Parser.jj
Comment (JiajunBernoulli) [reply core/src/main/codegen/templates/Parser.jj:4362]: Good catch.

File: server/src/main/java/org/apache/calcite/server/ServerDdlExecutor.java
Comment (JiajunBernoulli) [reply server/src/main/java/org/apache/calcite/server/ServerDdlExecutor.java:200]: Make sense, I added some explanations.

**Comment by ILuffZhe** [comment]
+1

**Review by ILuffZhe — APPROVED** [review]
+1. LGTM!

**Comment by JiajunBernoulli** [comment]
`rebase main` and `squash commits`, I will merge it tomorrow if no other suggestions.

**Pull request merged**
